### PR TITLE
Add hex_key parameter for IPMI input plugin connection

### LINKED ISSUE #6858: iDRAC 9 hex encryption Key input location
## Feature Request

Dell iDRAC 9 requires Encryption Key entry besides user and password for impi_sensor over LAN. Current plugin documentation only shows user and password? 

### Current behavior:
only user name and password entry example for ipmi_sensor output
### Desired behavior:
include entry example for encryption key
### Use case: 
working command for ipmitools requires -y for the encryption key, for example:
sudo ipmitool -I lanplus -H 192.168.1.1 -U User -P Password -y 000000000000000000000000000000000000000 sdr -v

### PR DESCRIPTION
**Pull request by Jagularr**
closes #6858

### CODE REVIEW

File: plugins/inputs/ipmi_sensor/connection.go [hunk 1]
```diff
@@ -15,11 +15,13 @@ type Connection struct {
...
 	Interface string
 	Privilege string
+	HexKey    string
 }
 
-func NewConnection(server string, privilege string) *Connection {
+func NewConnection(server, privilege, hexKey string) *Connection {
 	conn := &Connection{}
 	conn.Privilege = privilege
+	conn.HexKey = hexKey
```
Comment (srebhan) [inline plugins/inputs/ipmi_sensor/connection.go:24]: You can use
```suggestion
	conn := &Connection{
		Privilege: privilege,
		HexKey: hexKey,
	}
```
for initialization.

**Review by srebhan — APPROVED** [review]
A small (optional) suggestion in the code, otherwise this looks good. Is there a particular reason to substitute `assert.Equal` by `require.EqualValues` in the test-code?

File: plugins/inputs/ipmi_sensor/connection.go (hunk 1 above)
Comment (Jagularr) [reply plugins/inputs/ipmi_sensor/connection.go:24]: Good point, missed that.

**Comment by Jagularr** [comment]
Kind of just did that because I read [here](https://github.com/influxdata/telegraf/wiki/Review) that 'require' is preferred. Very possible that I misunderstood though.

**Comment by srebhan** [comment]
Awesome. Any thoughts on the cosmetic `Connection` init thingy? We can also keep as-is... Your choice!

**Comment by srebhan** [comment]
Oh and can you please rebase this again to trigger CI!? Running only 2 tests seems a bit strange...

**Comment by Jagularr** [comment]
Changed it and done!

**Review by srebhan — APPROVED** [review]
Perfect thanks!

**Review by helenosheaa — APPROVED** [review]
Looks good!

**Pull request merged**